tests/library/global-fetch.spec.ts

```diff
@@ -536,3 +536,47 @@ it('should retry ECONNRESET', {
   expect(requestCount).toBe(4);
   await request.dispose();
 });
+
+it('should throw when failOnStatusCode is set to true', async ({ playwright, server }) => {
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/34204' });
+  const request = await playwright.request.newContext({ apiRequest: { failOnStatusCode: true } });
+  server.setRoute('/empty.html', (req, res) => {
+    res.writeHead(404, { 'Content-Length': 10, 'Content-Type': 'text/plain' });
+    res.end('Not found.');
+  });
+  const error = await request.fetch(server.EMPTY_PAGE).catch(e => e);
+  expect(error.message).toContain('404 Not Found');
+});
+
+it('should throw when failOnStatusCode is set to true inside the fetch API call', async ({ playwright, server }) => {
```

Comment:
There is already `...should support failOnStatusCode` tests  that cover this scenario, no need to repeat.